### PR TITLE
Fixing #105, #104, #100, and #92

### PR DESCRIPTION
**Pull request by jessefeinman**
Fixed errors associated with bad responses/unexpected responses from tumblr.

Fixed infinite looping due to errors from bad responses. 

Print messages now contain the blog that they are from.

### CODE REVIEW

**Comment by CorentinB** [comment]
Thanks for your contribution bro! I just saw this project yesterday, but it's full of bugs, looks like the maintainer doesn't keep it updated anymore, but it's not a problem because the community is here!
Hope this PR will be merged soon. 😄

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: EOF sign is missing.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py:69]: I don't think we need to add `site` here. Actually `post` has already contains all the needed data, including links and etc.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py:194]: I think I've added retrying only for downloading posts.

For this kind of requests, there are no retrys currently. Thanks for fixing this.

**Review by dixudx — CHANGES_REQUESTED** [review]
@jessefeinman Thanks so much for your contribution. We need some changes before we get it merged.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py:240]: Duplicate codes here. We only need to have this retry logic in one place.

File: tumblr-photo-video-ripper.py
Comment (jessefeinman) [reply tumblr-photo-video-ripper.py:69]: It's a lot easier to follow by just threading through the site info into the queue entry then untuple it here to access it. Otherwise its not clear where the site info comes from due to the lack of explicit types on python.

File: tumblr-photo-video-ripper.py
Comment (jessefeinman) [reply tumblr-photo-video-ripper.py:194]: This is retry logic for each tumblr blog. This addresses the infinite loop issue where tumblr gives back an unexpected response so we can't parse it correctly. If that happens RETRIES number of times in a row then we stop trying to access that blog and move on to the next one. The infinite loop issue that you had before was the result of continuing the loop without advancing in the event of an exception.

Here we apply the same retry logic that we do for individual posts, ie. if we fail to parse the data returned by tumblr RETRIES times then we move on to the next url for that blog, if that again fails RETRIES times then we move on to the next site.

File: tumblr-photo-video-ripper.py
Comment (jessefeinman) [reply tumblr-photo-video-ripper.py:240]: Moved the retry logic to a single function.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: it seems the imported packages are not changed. No need to change here.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [reply tumblr-photo-video-ripper.py:69]: I'm okay with this.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: Try for `print`?

It does not make sense to have try/except for `print`.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: It does not make sense to have try/except for print.

File: tumblr-photo-video-ripper.py
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: This except is the same with except `UnicodeDecodeError`.

**Review by dixudx — CHANGES_REQUESTED** [review]
Some nits. We're almost ready to go.
Please also squash your commits.
Thanks so much for improving this. Awesome!!!

File: tumblr-photo-video-ripper.py
Comment (jessefeinman) [reply tumblr-photo-video-ripper.py]: media_url may not be defined if the exception was thrown before then. The try here is to give as much info as possible, so try and print out the media url but if its not available then just use what info is available

File: tumblr-photo-video-ripper.py
Comment (jessefeinman) [reply tumblr-photo-video-ripper.py]: The unicodedecodeerror is a known error that can occur so the message that is printed is more explicit about why it failed

Comment (dixudx) [reply tumblr-photo-video-ripper.py]: We can raise an exception in `unicodedecodeerror`. And print that error message in the last except.

**Review by dixudx — COMMENTED** [review]
@jessefeinman Please rebase and squash your commits. We're getting close.

**Pull request closed without merging**
